### DISCUSSION TITLE
One entity type not being persisted (others are OK)

### DISCUSSION THREAD

**Discussion by stefaanMLB** [mikro-orm/mikro-orm#2744] [Q&A]
**Describe the bug**
In a Nestjs application with a range of different entities, one of the entity types is not persisted to the database when creating a new record.  The other entity types work fine.
No INSERT instruction is being output to console (debug is on) or being sent to the database (watching general db log).
The same EntityRepository is being used successfully (just before the creation attempt) to query the same table and this returns correct results.

How can I further investigate this ? 

**code - entity definition**
```ts
import { Entity, PrimaryKey, Property } from '@mikro-orm/core'

@Entity({ tableName: 'uitpak_counters' })
export class UitpakCounters {
  @PrimaryKey() 
  id: number | null = null

  @Property({ length: 30, nullable: false })
  key!: string

  @Property({ length: 30, nullable: true })
  modifier: string | null = null

  @Property({ nullable: true, default: null })
  value: number | null = null
}
```

**code - entity insertions (not working)**
```ts
  private async addCounter(key: string, modifier: string | null, value: number): Promise<Either<Error, void>> {
    try {
      const newCounter = this._counterRepository.create({ key, modifier, value })
      await this._counterRepository.persistAndFlush(newCounter)
      return Right(undefined)
    } catch (error) {
      //...
    }
  }
```

**code - other use of the same EntityRepository (works fine)**
```ts
  async checkCounterExists(orderType: OrderType, monthIndicator: string): Promise<boolean> {
    try {
      const modifier = TRANSLATION_TABLE[orderType] + monthIndicator
      const count = await this._counterRepository.count({ key: ORDER_NUMBER_KEY, modifier })
      return count > 0
    } catch (error) {
      //...
    }
  }
```

**code - other entity insertion that works fine**
```ts
  addTubeActionLog(log: TubeLogAction): void {
    //...

    const tubeLog = this._tubesLoggingRepo.create({
      requestnumber: log.request_number,
      ...
    })

    this._orm.em.persistAndFlush([tubeLog])
  }
```

**Versions**
    "@mikro-orm/core": "^4.5.9",
    "@mikro-orm/mariadb": "^4.5.9",
    "typescript": "^4.3.5"
  "mikro-orm": {
    "debug": true,
    "useTsNode": true,
    "configPaths": [
      "./mikro-orm.config.ts",
      "./mikro-orm.config.js"
    ]
  },

**Comment by B4nan** [comment]
This is very incomplete, if you want help, please provide complete but minimal reproduction, as a code that I can execute. Also try upgrading to v5 (or at least 4.5.10, but dont expect any fixes to v4).

Note that in your last snippet you are missing `await` on the flush call.

**Comment by stefaanMLB** [comment]
@B4nan, I am converting the project to mikro-orm v5 but get stuck on the `Loaded<>` type that is returned from the `.find()` functions.
I need to pass some entities into functions that converts them into DTO objects.  The `Loaded<SomeEntity,never>` type returned by the find function is not compatible with my `SomeEntity` type and the `Loaded` type doesn't seem to offer a way to convert the object into the related entity.
How can I convert an object of `Loaded<SomeEntity, never>` into the related `SomeEntity` type ?

**Reply by B4nan** [reply]
its a type, you dont have to convert anything, just type cast

**Reply by stefaanMLB** [reply]
OK, yes.  Sorry for the dumb question.  I had a different error that turned the cast into an error.
